### PR TITLE
Risc-V Linux port for TensorFlow Lite

### PR DESCRIPTION
**Pull request by vaibhavexleapsemi**
This PR has addition for riscv_makefile.inc
Pre-requistes:
- Install RISC-V GNU Toolchain for Linux. Refer https://github.com/riscv/riscv-gnu-toolchain
- Here we are using riscv64-unknown-linux-gnu- tools created for "linux" from above link.
The changes builds executables and static libs  in "linux_riscv64", in "gen" directory.
Tested on RISC-V HiFive Unleashed.
Test Results on RISC-V HiFive Unleashed:
$./label_image --tflite_model ./models/mobilenet_v1_1.0_224.tflite --labels labels.txt -i ./images/grace_hopper.bmp 
INFO: Loaded model ./models/mobilenet_v1_1.0_224.tflite
INFO: resolved reporter
INFO: invoked
INFO: average time: 1553.19 ms
INFO: 0.860174: 653 653:military uniform
INFO: 0.0481021: 907 907:Windsor tie
INFO: 0.00786705: 466 466:bulletproof vest
INFO: 0.00644936: 514 514:cornet, horn, trumpet, trump
INFO: 0.00608029: 543 543:drumstick

### CODE REVIEW

File: tensorflow/lite/tools/make/build_riscv_lib.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+FREE_MEM="$(free -m | awk '/^Mem/ {print $2}')"
+# Use "-j 4" only memory is larger than 2GB
+if [[ "FREE_MEM" -gt "2000" ]]; then
+  NO_JOB=4
+else
+  NO_JOB=1
+fi
+
+make -j ${NO_JOB} TARGET=linux_riscv64 -C "${TENSORFLOW_DIR}" -f tensorflow/lite/tools/make/Makefile $@
```
Comment (terryheo) [inline tensorflow/lite/tools/make/build_riscv_lib.sh:31]: Don't you need to provide a way to build 32bit build?

File: tensorflow/lite/tools/make/build_riscv_lib.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash 
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
```
Comment (terryheo) [inline tensorflow/lite/tools/make/build_riscv_lib.sh:2]: 2020?

File: tensorflow/lite/tools/make/build_riscv_lib.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# ==============================================================================
+
+set -x
+set -e
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+TENSORFLOW_DIR="${SCRIPT_DIR}/../../../.."
+
+FREE_MEM="$(free -m | awk '/^Mem/ {print $2}')"
+# Use "-j 4" only memory is larger than 2GB
```
Comment (terryheo) [inline tensorflow/lite/tools/make/build_riscv_lib.sh:24]: I needed this to use it on RPI3. I think most desktops are fine to just use "make -j" without giving a job numer?

File: tensorflow/lite/tools/make/build_riscv_lib.sh (hunk 2 above)
Comment (vaibhavexleapsemi) [reply tensorflow/lite/tools/make/build_riscv_lib.sh:2]: @terryheo HI,
Will change.

Thanks

File: tensorflow/lite/tools/make/build_riscv_lib.sh (hunk 3 above)
Comment (vaibhavexleapsemi) [reply tensorflow/lite/tools/make/build_riscv_lib.sh:24]: You are right. But on my side I am working on freedom-u540 which has 4 core. It is some what similar to raspberry pi in terms of number of cores. Hence I kept the script same.
For more details I am providing a link on freedom-u540 https://www.sifive.com/boards/hifive-unleashed
Please let us know if you need additional info.

File: tensorflow/lite/tools/make/build_riscv_lib.sh (hunk 1 above)
Comment (vaibhavexleapsemi) [reply tensorflow/lite/tools/make/build_riscv_lib.sh:31]: > Don't you need to provide a way to build 32bit build?

Actually, we tried on HiFive1 (32 bit MCU). I am getting issues with it.
Once it is fixed , will add support for RISC-V 32 bit MCU.

File: tensorflow/lite/tools/make/build_riscv_lib.sh (hunk 3 above)
Comment (vaibhavexleapsemi) [reply tensorflow/lite/tools/make/build_riscv_lib.sh:24]: @terryheo Can you suggest what is needed from my side.

Comment (terryheo) [reply tensorflow/lite/tools/make/build_riscv_lib.sh:24]: If that's the case, it's fine to keep the logic.

File: tensorflow/lite/tools/make/build_riscv_lib.sh (hunk 1 above)
Comment (terryheo) [reply tensorflow/lite/tools/make/build_riscv_lib.sh:31]: I see. I think we might need to use an overridable variable to control this. You can do it later once it's ready.

**Review by terryheo — APPROVED** [review]

**Comment by vaibhavexleapsemi** [comment]
@terryheo Can you please tell me, during checks why it is failing for **"import/copybara — An error happened while migrating the change".**
 Is it because of outdated license? (tensorflow/lite/tools/make/build_riscv_lib.sh Outdated )

File: tensorflow/lite/tools/make/build_riscv_lib.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (terryheo) [inline tensorflow/lite/tools/make/build_riscv_lib.sh:1]: there is an unnecessary character after 'bash'. plz remove it.

**Comment by terryheo** [comment]
I'll handle the copybara error. In the meantime, could you address my comment?

File: tensorflow/lite/tools/make/build_riscv_lib.sh (hunk 4 above)
Comment (vaibhavexleapsemi) [reply tensorflow/lite/tools/make/build_riscv_lib.sh]: Yes, Sure.

**Comment by vaibhavexleapsemi** [comment]
> I'll handle the copybara error. In the meantime, could you address my comment?

As you suggested I had removed unwanted character(space) after #!/bin/bash

**Review by terryheo — APPROVED** [review]

**Pull request merged**